xbmc/video/VideoDatabase.cpp

```diff
@@ -8708,8 +8723,9 @@ std::vector<int> CVideoDatabase::CleanMediaType(const std::string &mediaType, co
                !sourcePathsDeleteDecision->second.second)
         del = false;
 
+      // delete parents folder (and all its items) only, if file needs to be deleted and folder does not exist
       if (scanSettings.parent_name)
-        pathsDeleteDecisions.insert(std::make_pair(m_pDS2->fv(2).get_asInt(), del));
+        pathsDeleteDecisions.insert(std::make_pair(m_pDS2->fv(2).get_asInt(), !CDirectory::Exists( parentPath, false ) && del));
```

Comment:
This extra call to `CDirectory::Exists()` might bring back a regression we had in the past where we try to check a path that is not online which takes a lot of time. That's the main reason why we do all this caching in `pathsDeleteDecisions` and `sourcePathsDeleteDecisions`. But it's very hard to say just from reading the code. It certainly doesn't make any sense to check the parent path's existance if we have already figured out that the source the path belongs to doesn't exist anymore.